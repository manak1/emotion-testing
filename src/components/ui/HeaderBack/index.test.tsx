import { render, screen } from "~/test/test-utils"
import * as stories from "./index.stories"
import { composeStories } from "@storybook/testing-react"
const { Default } = composeStories(stories)

describe("HeaderBackのテスト", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Default />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("titleが表示されている", () => {
    render(<Default />)
    expect(screen.getByText("タイトル")).toBeInTheDocument()
  })
})
