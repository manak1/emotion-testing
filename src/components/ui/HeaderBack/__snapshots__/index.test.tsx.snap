// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackのテスト snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  border: 1px solid #d3d3d3;
  max-width: 480px;
  margin: 0 auto;
  padding: 8px 0;
}

.emotion-1 {
  background-color: #fff;
  position: relative;
  width: 100%;
  max-width: 480px;
  padding: 0 16px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  width: 56px;
  height: 56px;
  border-radius: 9999px;
  position: relative;
  outline: none;
  border: none;
  cursor: pointer;
  background-color: transparent;
  -webkit-transition: box-shadow 0.2s ease;
  transition: box-shadow 0.2s ease;
  width: 34px;
  height: 34px;
}

.emotion-2:focus {
  box-shadow: 0 0 0 2px rgba(120, 86, 255, 0.3);
  -webkit-transition: box-shadow 0.2s ease-out;
  transition: box-shadow 0.2s ease-out;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  position: absolute;
  top: 8px;
  left: 8px;
  color: #333;
  width: 24px;
  height: 24px;
  top: 5px;
  left: 5px;
}

.emotion-5 {
  color: #333;
  font-weight: bold;
  font-size: 20px;
  margin: 0 0 0 16px;
}

<header
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <button
        class="emotion-2"
      >
        <svg
          aria-hidden="true"
          class="emotion-3 emotion-4"
          fill="currentColor"
          focusable="false"
          height="20px"
          variant="transparent"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </button>
      <h1
        class="emotion-5"
      >
        タイトル
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
